Assignment2/Logger/LogFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace Logger
+{
+    
+  
+    public class LogFactory
+    {
+        private string LogFilePath { get; set; }
+        public BaseLogger CreateLogger(string className)
+        {
+            if (!string.IsNullOrEmpty(LogFilePath))
+            {
+                BaseLogger baseLogger = new FileLogger(LogFilePath);
+                baseLogger.ClassName = className;
```

Comment:
While this still works, it doesn't meet the requirements of the writeup. The requirement was to use an **object initializer** which allows you to construct the object and assign `Classname` at the same time. Check out [these Microsoft docs for more details](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/how-to-initialize-objects-by-using-an-object-initializer)

```suggestion
                BaseLogger baseLogger = new FileLogger(LogFilePath) { Classname = classname }
```